src/fifo/FifoBuffer.cpp

```diff
@@ -99,28 +99,26 @@ int32_t FifoBuffer::read(void *buffer, int32_t numFrames) {
     uint8_t *destination = reinterpret_cast<uint8_t *>(buffer);
     uint8_t *source = &mStorage[convertFramesToBytes(readIndex)];
     if ((readIndex + framesToRead) > mFrameCapacity) {
-        // read in two parts, first part here is at the end of the mStorage buffer
-        uint32_t frames1 = mFrameCapacity - readIndex;
-        int32_t numBytes = convertFramesToBytes(frames1);
-        if (numBytes < 0) {
+        // First, read frames until the end of the mStorage buffer.
+        if(readIndex > mFrameCapacity) {
```

Comment:
It seems like this should never happen, if we trust mFifo to be internally consistent. Should this check still be here?